src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructure.java

```diff
@@ -0,0 +1,286 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class computes a number of useful structural properties for DAG specified by the
+ * immediate parents and immediate children of each node.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ComputableGraphStructure {
```

Comment:
You are right. Most of them are used in the constructor and since the seemed useful things to have around, I wrote getters for them as well. I will leave them for now. We'll get to refactoring ICG later. BTW, I cleaned up the ICG code significantly following yours and David's comments.